### ISSUE TITLE
Failure building with pip from source

### ISSUE THREAD

**Issue by willrogers** [mdavidsaver/p4p#20]
This happened because I was using Python 3.7 and there was no pre-built wheel.

```
(venv3) hgs15624@pc0072 ~/t/p4p (master) [1]> pip install epicscorelibs
Collecting epicscorelibs
  Using cached https://files.pythonhosted.org/packages/34/2f/bcc5bdebafa94d02efea46cc710547775e6a4a48fa30adaafe453dabd32f/epicscorelibs-7.0.1.99.15.1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-1v_5ltn5/epicscorelibs/setup.py", line 13, in <module>
        from setuptools_dso import setup, DSO, build_dso
    ModuleNotFoundError: No module named 'setuptools_dso'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-1v_5ltn5/epicscorelibs/
```

**Comment by mdavidsaver** [comment]
For the moment you'll have to install the deps piecewise.

```
python -m pip install setuptools_dso nose numpy
python -m pip install --no-binary :all: epicscorelibs
python -m pip install --no-binary :all: p4p
python -m nose p4p
```

I'm unsure how to make this process smoother without duplicating the logic in `setuptools_dso` in both p4p and epicscorelibs, or resorting to ugly hacks along the lines of `if 'egg_info' in sys.argv`.

I did some experiments using the (very) new `pyproject.toml` (aka PEP518) to capture dependencies of setup.py, but couldn't make this work either.

There is the `setup_depends` key for setuptools.  This does work, but I found it to be extremely annoying that `setup.py -h` would hit the network, and install things.
